src/main/java/org/folio/rmapi/PackageServiceImpl.java

```diff
@@ -25,15 +27,31 @@
 
   private ProvidersServiceImpl providerService;
   private TitlesHoldingsIQService titlesService;
+  private VertxCache<PackageCacheKey, PackageByIdData> packageCache;
+  private Configuration configuration;
+  private String tenantId;
 
-  public PackageServiceImpl(Configuration config, Vertx vertx, ProvidersServiceImpl providerService, TitlesHoldingsIQService titlesService) {
+  public PackageServiceImpl(Configuration config, Vertx vertx, String tenantId, ProvidersServiceImpl providerService,
+                            TitlesHoldingsIQService titlesService, VertxCache<PackageCacheKey, PackageByIdData> packageCache) {
     super(config, vertx);
     this.providerService = providerService;
     this.titlesService = titlesService;
+    this.packageCache = packageCache;
+    this.tenantId = tenantId;
+    this.configuration = config;
   }
 
   public CompletableFuture<PackageResult> retrievePackage(PackageId packageId, List<String> includedObjects) {
-    CompletableFuture<PackageByIdData> packageFuture = retrievePackage(packageId);
+    return retrievePackage(packageId, includedObjects, false);
+  }
+
+  public CompletableFuture<PackageResult> retrievePackage(PackageId packageId, List<String> includedObjects, boolean useCache) {
+    CompletableFuture<PackageByIdData> packageFuture;
+    if(useCache){
+      packageFuture = retrievePackageWithCache(packageId);
```

Comment:
can we rename this to retrievePackageFromCache instead? Thanks!

Reply:
I'm not sure, to me retrievePackageFromCache would mean that it returns null on cache miss, while this method loads package from RM API on cache miss

Reply:
Hmm - ok - just a naming thing. guess this is ok.